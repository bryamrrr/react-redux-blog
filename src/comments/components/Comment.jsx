import React, { PropTypes } from 'react';
import { FormattedHTMLMessage } from 'react-intl';

function Comment(props) {
  return (
    <article id={`comment-${props.id}`}>
      <div>
        <FormattedHTMLMessage
          id="comment.meta.author"
          values={{
            name: props.name,
            email: props.email,
          }}
        />
      </div>

      <p>
        {props.body}
      </p>
    </article>
  );
}

Comment.propTypes = {
  id: PropTypes.number.isRequired,
  email: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
};

export default Comment;
