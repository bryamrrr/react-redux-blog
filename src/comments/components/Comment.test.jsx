import React from'react';
import renderer from'react-test-renderer';
import { IntlProvider } from'react-intl';

import Comment from'./Comment.jsx';

import messages from'../../messages.json';

const comment = {
  id: 1,
  email: 'bryam@gmail.com',
  name: 'Bryam Rodriguez',
  body: 'Este es un comentario de prueba',
};

test('Comment should render the component', () => {
  const tree = renderer.create(
    <IntlProvider locale="es" messages={messages.es}><Comment {...comment} /></IntlProvider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});