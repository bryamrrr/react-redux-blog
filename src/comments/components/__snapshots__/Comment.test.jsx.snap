// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment should render the component 1`] = `
<article
  id="comment-1"
>
  <div>
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "Por: <a href=\\"mailto:bryam@gmail.com\\" target=\\"_blank\\">Bryam Rodriguez</a>",
        }
      }
    />
  </div>
  <p>
    Este es un comentario de prueba
  </p>
</article>
`;
